src/exchange/services/exchange.order.service.js

```diff
@@ -0,0 +1,190 @@
+
+        this.exchangeServiceInfrastructure = exchangeServiceInfrastructure;
+        this.exchangeVersion = exchangeVersion;
+        this.OvhApiMsServices = OvhApiMsServices;
+        this.OvhApiOrder = OvhApiOrder;
+        this.User = User;
+      }
+
+      fetchingDoesServiceUseAgora(serviceName) {
+        return this.OvhApiMsServices.Exchange().v6()
+          .doesServiceUseAgora({ serviceName }).$promise
+          .then(({ serviceUsesAgora }) => serviceUsesAgora);
+      }
+
+      creatingCart() {
+        return this.User
+          .getUser()
+          .then(({ ovhSubsidiary }) => this.OvhApiOrder.Cart().v6()
+            .post({ ovhSubsidiary }).$promise)
+          .then(({ cartId }) => this.OvhApiOrder.Cart().v6().assign({ cartId }).$promise)
+          .then(({ cartId }) => cartId);
+      }
+
+      fetchingProductOffers() {
+        const serviceInfrastructureName = this.exchangeServiceInfrastructure
+          .getName()
+          .toLowerCase();
+        const serviceVersion = this.exchangeVersion.get();
+
+        const localOffers = [
+          {
+            orderOfAppearance: 1,
+            planCode: `exchange-account-${serviceInfrastructureName}-${serviceVersion}`,
+            valueToDisplay: this.$translate.instant(`exchange_account_order_accountType_item_exchange-account-${serviceInfrastructureName}-${serviceVersion}`),
+            prices: {
+              P1M: {
+                duration: 'P1M',
+              },
+              P1Y: {
+                duration: 'P1Y',
+              },
+            },
+          },
+          {
+            orderOfAppearance: 2,
+            planCode: `exchange-account-${serviceInfrastructureName}-${serviceVersion}-enterprise`,
+            valueToDisplay: this.$translate.instant(`exchange_account_order_accountType_item_exchange-account-${serviceInfrastructureName}-${serviceVersion}-enterprise`),
+            prices: {
+              P1M: {
+                duration: 'P1M',
+              },
+              P1Y: {
+                duration: 'P1Y',
+              },
+            },
+          },
+          {
+            orderOfAppearance: 3,
+            planCode: `exchange-account-${serviceInfrastructureName}-${serviceVersion}-300g`,
+            valueToDisplay: this.$translate.instant(`exchange_account_order_accountType_item_exchange-account-${serviceInfrastructureName}-${serviceVersion}-300g`),
+            prices: {
+              P1M: {
+                duration: 'P1M',
+              },
+              P1Y: {
+                duration: 'P1Y',
+              },
+            },
+          },
+          {
+            orderOfAppearance: 4,
+            planCode: `exchange-account-${serviceInfrastructureName}-${serviceVersion}-300g-enterprise`,
+            valueToDisplay: this.$translate.instant(`exchange_account_order_accountType_item_exchange-account-${serviceInfrastructureName}-${serviceVersion}-300g-enterprise`),
+            prices: {
```

Comment:
Maybe `prices` could be extracted as a `const` and reused for each offer

Reply:
Yes it could